### DISCUSSION TITLE
imports for plugins

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [streamlink/streamlink#3979] [Q&A]
I am going to write a plugin to handle streams from a certain website.

The site uses javascript and (as far as I know) to get the stream urls I need to use selenium, seleniumwire, and geckodriver. I am able to get the stream urls already in an outside script. How does one go about including modules with a plugin script?

**Comment by gravyboat** [comment]
We won't support adding any site to the application that would require so many dependencies. You're welcome to fork or [sideload](https://streamlink.github.io/cli.html?highlight=side%20load#sideloading-plugins). You may want to mention what the site is here if it meets the plugin requirements, others may have a better way of doing this.

**Reply by ReenigneArcher** [reply]
Okay thank you. I guess for sideloaded plugins I would just put the dependencies next to my plugin script. Correct?

The issue is that the video must played, and await the url of the m3u8 playlist which also contains the access token. The token is only valid for 4 hours, and it is also tied to the ip address that originally requested it. This causes a problem if you want to use a proxy to view the streams because the token will not work for the new ip address.

Unfortunately the link or token cannot be gotten simply by viewing the page source or using any api call.
